infra/Pos/Binary/Infra/Slotting.hs

```diff
@@ -6,26 +6,26 @@ module Pos.Binary.Infra.Slotting
 
 import           Universum
 
-import           Pos.Binary.Class   (Bi (..))
+import           Pos.Binary.Class   (Bi (..), label, labelS, putField)
 import           Pos.Binary.Core    ()
 import           Pos.Slotting.Types (EpochSlottingData (..), SlottingData (..))
 
 instance Bi EpochSlottingData where
-    put EpochSlottingData {..} = do
-        put esdSlotDuration
-        put esdStart
-    get = do
+    sizeNPut = labelS "EpochSlottingData" $
+        putField esdSlotDuration <>
+        putField esdStart
+    get = label "EpochSlottingData" $ do
         esdSlotDuration <- get
-        esdStart <- get
-        return EpochSlottingData {..}
+        esdStart        <- get
+        pure EpochSlottingData{..}
 
 instance Bi SlottingData where
-    put SlottingData {..} = do
-        put sdPenult
-        put sdLast
-        put sdPenultEpoch
-    get = do
-        sdPenult <- get
-        sdLast <- get
+    sizeNPut = labelS "SlottingData" $
+        putField sdPenult <>
+        putField sdLast <>
+        putField sdPenult
+    get = label "SlottingData" $ do
+        sdPenult      <- get
+        sdLast        <- get
         sdPenultEpoch <- get
-        return SlottingData {..}
+        pure SlottingData{..}
```

Comment:
TH doesn't work with parameterized type (or types with parametrized types, like
`data A = {
    field :: Maybe Int
}`
, it will be fixed in the next PR